source/common/stats/allocator_impl.h

```diff
@@ -34,7 +34,21 @@ class AllocatorImpl : public Allocator {
   void debugPrint();
 #endif
 
+  /**
+   * @return a thread synchronizer object used for reproducing a race-condition in tests.
+   */
+  Thread::ThreadSynchronizer& sync() { return sync_; }
+
+  /**
+   * @return whether the allocator's mutex is locked, exposed for testing purposes.
+   */
+  bool isMutexLocked();
```

Comment:
It would be helpful to add "ForTest" to the method name to make the intent more clear.